### PR TITLE
Adding matlab reading capabilities for new format in CellExplorerExtractor

### PR DESCRIPTION
**Pull request by h-mayorquin**
The purpose of this pull request is to fix/update some reading problems with the reading of matlab files in the `CellExplorerExtractor` 

1) We add capabilities to consistently read `sessionInfo` and and `spikes.Info` files independently on the matlab format (that is either `scipy.io` or `hdf5storage`).
2) Add the location of the the matlab files as an optional parameter in the extractor.
3) Add the sampling frequency as an optional parameter so a reading of `SessionInfo` is not required.

### CODE REVIEW

File: spikeextractors/extractors/cellexplorersortingextractor/cellexplorersortingextractor.py [hunk 1]
```diff
@@ -26,41 +29,78 @@ class CellExplorerSortingExtractor(SortingExtractor):
...
+        ), f"The spikes_matfile_path ({spikes_matfile_path}) must exist!"
         folder_path = spikes_matfile_path.parent
         sorting_id = spikes_matfile_path.name.split(".")[0]
+        if sampling_frequency is None:
+            if session_info_matfile_path is None:
+                session_info_matfile_path = folder_path / f"{sorting_id}.sessionInfo.mat"
+
+            assert (
+                session_info_matfile_path.is_file()
+            ), "No sessionInfo.mat file found in the folder!"
```
Comment (alejoe91) [inline spikeextractors/extractors/cellexplorersortingextractor/cellexplorersortingextractor.py:53]:
```suggestion
            ), f"No {sorting_id}.sessionInfo.mat file found in the folder!" 
```

File: spikeextractors/extractors/cellexplorersortingextractor/cellexplorersortingextractor.py [hunk 2]
```diff
@@ -26,41 +29,78 @@ class CellExplorerSortingExtractor(SortingExtractor):
...
+            if self.read_session_info_with_scipy:
+                self._sampling_frequency = float(
+                    session_info_mat["sessionInfo"]["rates"][0][0]["wideband"][0][0][0][0]
+                )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
+            else:
+                self._sampling_frequency = float(
+                    session_info_mat["sessionInfo"]["rates"][0][0]["wideband"][0][0]
+                )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
+        else:
+            self._sampling_frequency = sampling_frequency
```
Comment (alejoe91) [inline spikeextractors/extractors/cellexplorersortingextractor/cellexplorersortingextractor.py:76]:
```suggestion
            assert sampling_frequency is not None, "Could not load sampling frequency from files. Use the 'sampling_frequency' argument"
            self._sampling_frequency = sampling_frequency
```

**Review by alejoe91 — CHANGES_REQUESTED** [review]
@h-mayorquin Thanks for these fixes! I had just a couple of small comments

File: spikeextractors/extractors/cellexplorersortingextractor/cellexplorersortingextractor.py [hunk 3]
```diff
@@ -26,41 +29,79 @@ class CellExplorerSortingExtractor(SortingExtractor):
...
+            if self.read_session_info_with_scipy:
+                self._sampling_frequency = float(
+                    session_info_mat["sessionInfo"]["rates"][0][0]["wideband"][0][0][0][0]
+                )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
+            else:
+                self._sampling_frequency = float(
+                    session_info_mat["sessionInfo"]["rates"][0][0]["wideband"][0][0]
+                )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
+        else:
+            assert sampling_frequency is not None, "Could not load sampling frequency from files. Use the 'sampling_frequency' argument"
```
Comment (h-mayorquin) [inline spikeextractors/extractors/cellexplorersortingextractor/cellexplorersortingextractor.py:76]: @alejoe91  you suggested this assert but I am confused about it. The program only enters this else when the sampling_frequency is not None, so why check for it again?

**Review by h-mayorquin — COMMENTED** [review]
I removed some comments and formatting issues (I have to say there are some aspects of black here that I don't like but I kept them that way).

**Comment by h-mayorquin** [comment]
@alejoe91 what else would be required for this to be merged?

File: spikeextractors/extractors/cellexplorersortingextractor/cellexplorersortingextractor.py (hunk 3 above)
Comment (alejoe91) [reply spikeextractors/extractors/cellexplorersortingextractor/cellexplorersortingextractor.py:76]: Right, you can remove this else ;)

**Comment by h-mayorquin** [comment]
> > @alejoe91 what else would be required for this to be merged?
> 
> Looks good! Waiting for tests and merging.

All right, removed the assert. It seems that the tests were failing tough. I will check that out.

**Comment by alejoe91** [comment]
> All right, removed the assert. It seems that the tests were failing tough. I will check that out.

Don't worry about it. It seems some weird dependency issue with tridesclous. I'll fix it ;)

**Pull request merged**